herddb-core/src/test/java/herddb/data/consistency/MultiNodeConsistencyCheckTest.java

```diff
@@ -0,0 +1,92 @@
+import herddb.model.TransactionContext;
+import herddb.model.commands.CreateTableSpaceStatement;
+import herddb.model.commands.CreateTableStatement;
+import herddb.model.commands.TableConsistencyCheckStatement;
+import herddb.utils.SystemInstrumentation;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.Test;
+
+
+/**
+ *
+ * @author hamado
+ */
+public class MultiNodeConsistencyCheckTest extends ReplicatedLogtestcase {
+
+    final String tableName = "table1";
+    final String tableSpaceName = "t2";
+
+    @Test
+    public void consistencyCheckReplicaTest() throws Exception {
+        final AtomicInteger callCount = new AtomicInteger();
+        SystemInstrumentation.addListener(new SystemInstrumentation.SingleInstrumentationPointListener("createChecksum") {
+            @Override
+            public void acceptSingle(Object... args) throws Exception {
+                callCount.incrementAndGet();
+            }
+        });
+        try (DBManager manager1 = startDBManager("node1")) {
+            manager1.executeStatement(new CreateTableSpaceStatement(tableSpaceName, new HashSet<>(Arrays.asList("node1", "node2")), "node1", 2, 0, 0), StatementEvaluationContext.DEFAULT_EVALUATION_CONTEXT(), TransactionContext.NO_TRANSACTION);
+            manager1.waitForTablespace(tableSpaceName, 10000, true);
+            try (DBManager manager2 = startDBManager("node2")) {
+                manager2.waitForTablespace(tableSpaceName, 10000, false);
+                Table table = Table
+                    .builder()
+                    .tablespace(tableSpaceName)
+                    .name(tableName)
+                    .column("id", ColumnTypes.STRING)
+                    .column("name", ColumnTypes.BOOLEAN)
+                    .primaryKey("id")
+                    .build();
+
+                CreateTableStatement st = new CreateTableStatement(table);
+                manager1.executeStatement(st, StatementEvaluationContext.DEFAULT_EVALUATION_CONTEXT(), TransactionContext.NO_TRANSACTION);
+
+                manager1.waitForTable(tableSpaceName, tableName, 10000, true);
+                execute(manager1, "INSERT INTO t2.table1 (id,name) values (?,?)", Arrays.asList("1", true));
+                execute(manager1, "INSERT INTO t2.table1 (id,name) values (?,?)", Arrays.asList("2", false));
+                manager2.waitForTable(tableSpaceName, tableName, 10000, false);
+
+                TableConsistencyCheckStatement statement = new TableConsistencyCheckStatement("table1", "t2");
+                DataConsistencyStatementResult result = manager1.createTableCheckSum(statement, null);
+                assertTrue("Consistency check replicated test ", result.getOk());
+
+                //The follower is always back 1, I make an entry to make him apply consistency log
+                execute(manager1, "INSERT INTO t2.table1 (id,name) values (?,?)", Arrays.asList("3", false));
```

Comment:
@eolivelli 
I saw that the follower always lags 1 when applying the logs, so to make him apply the consistency entry I had to make another insert. Is this normal behavior?

Reply:
Yes.
It is normal